### PR TITLE
3249 - Add more short form sizes and a form level toggle

### LINKED ISSUE #3249: Forms: Implement optional smaller field sizes
**Is your feature request related to a problem? Please describe.**

We need to implement a (design to follow) design that has smaller form fields and reduce field and padding sizes so that larger forms can fit in the pages better. This is from customer feedback upgrading from 3.x to 4.x.

The roughly means adding a `field-small-stacked` or `field-short-stacked` class like the current `field-short` but with new designed sizes and the labels on top. 

**Describe the solution you'd like**

- [x] All changes should go in tokens, so add field sizes and form layout details for normal, short fields and short fields stacked.
- [x] Add field-short-stacked example like field-short examples and test all fields and widths (Screen Shot 1)
- [ ] Reduce bottom padding on context forms (Screen Shot 2)
- [ ] Reduce bottom padding between data labels on summary fields  (Screen Shot 3)
- [x] Add some form label examples showing size between fields  (Screen Shot 4)
- [x] Add form level “switch” to toggle compressed mode in an example with a possible form level class and menu item
- [x] Make test page with all form features for testing

**Component List**
- [x]  autocomplete
- [x]  button
- [x]  checkbox
- [x]  dropdown
- [x]  colorpicker
- [x]  datepicker
- [x]  lookup
- [x]  timepicker
- [x]  radios
- [x]  spinbox
- [x]  completion-chart? (in landmark called progress bar)
- [x]  input (done)
- [x]  tags
- [x]  hyperlinks
- [x]  textarea
- [x]  fileupload
- [ ]  field filters
- [ ]  editor
- [ ]  fieldset/expandable area
- [ ]  rating
- [ ]  switch
- [ ]  slider
- [ ] context forms
- [ ] summary forms
- [ ] counts - would counts be smaller?
- [x]  badge/tag - i think the size should be ok for both personally
- [x]  toolbars A: No Change as no padding.
- [x]  input with validation messages
- [x]  paragraphs  - Q: `<p>` for text marked as "display as text" A: Use tokens `$input-size-regular-font-size, $label-size-regular-font-size, $input-size-compact-font-size, $label-size-compact-font-size`
- [x]  icons - Q: would these stay the same height? A: in fields they go down but buttons stay same size

**Describe alternatives you've considered**
N/A

**Additional context**
Outlook email "Our initial POC / mocks for "compact" form layouts"

**Screen Shot 1**
image001

**Screen Shot 2**
image002

**Screen Shot 3**
image003

**Screen Shot 4**
image004

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Updated the short form code to include more components and added a toggle so that short fields can be toggled with a form class.

**Related github/jira issue (required)**:
Fixes #3249 

**Steps necessary to review your pull request (required)**:
- Pull this branch and `npm i` and restart demo server
- test the following pages in RTL and both uplift  (vibrant) and soho (subtle) theme
- http://localhost:4000/components/input/test-short-field-sizes.html
- http://localhost:4000/components/input/test-short-fields-validated.html
- http://localhost:4000/components/input/test-short-fields.html
- on this page you can toggle the form class with a button on the right side. only the left side will toggle for comparison
- http://localhost:4000/components/form/example-compact-mode.html

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by tmcconechy** [comment]
Need to fix a few tests but posting for now.

**Comment by tmcconechy** [comment]
@deep7102 @EdwardCoyle This is ready to review. Sorry for the size of it.
Just fixing a few small things in tests that shouldnt effect it much

File: src/components/dropdown/_dropdown-uplift.scss [hunk 1]
```diff
@@ -75,15 +75,23 @@ div.multiselect {
 }
 
 // Adjust Short Field Dropdowns
-.field-short div.dropdown,
-.field-short div.multiselect {
-  padding: 2px 16px 2px 5px;
+.field-short,
+.form-layout-compact .field {
```
Comment (EdwardCoyle) [inline src/components/dropdown/_dropdown-uplift.scss]: Noticed this icon out of alignment on http://localhost:4000/components/form/example-compact-mode.html?theme=uplift&variant=light&locale=ar-EG:

Screen Shot 2020-01-15 at 2 38 00 PM

Maybe originating here?

File: app/views/components/form/example-compact-mode.html [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        <input type="text" id="example-text" name="example-text" value="Text"/>
+      </div>
+
+      <div class="field">
+        <label for="example-text-placeholder">Regular Placeholder</label>
+        <input type="text" id="example-text-placeholder" name="example-text-placeholder" placeholder="Useful Info"/>
+      </div>
+
+      <div class="field">
+        <label for="example-text-validation" class="required">Validation</label>
```
Comment (EdwardCoyle) [inline app/views/components/form/example-compact-mode.html]: Not sure if the alignment of the validation icons/messages is right on RTL:

Screen Shot 2020-01-15 at 2 40 03 PM

**Comment by deep7102** [comment]
All looks good, just found few things
http://localhost:4000/components/input/test-short-field-sizes.html?locale=ar-EG

- Button content alignment
- Multiselect list position (should extend extra width towards to left side)
    image
- Colorpicker size on mobile view
    image

**Comment by tmcconechy** [comment]
Will merge this and fix these issues on the beta.

**Pull request merged**